types/src/transaction/mod.rs

```diff
@@ -2640,15 +2640,70 @@ pub trait BlockExecutableTransaction: Sync + Send + Clone + 'static {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum ViewFunctionError {
+    // This is to represent errors are from a MoveAbort and has error info from the module metadata to display.
+    // The ExecutionStatus is used to construct the error message in the same way as MoveAborts for entry functions.
+    ExecutionStatus(ExecutionStatus, Option<StatusCode>),
```

Comment:
It is a bit confusing to have execution status and status code at the same time, but we can refactor this later if we need it, I guess :)